diff --git a/worker/src/RTC/SrtpSession.cpp b/worker/src/RTC/SrtpSession.cpp
index 97051e0cf..d87f03ae4 100644
--- a/worker/src/RTC/SrtpSession.cpp
+++ b/worker/src/RTC/SrtpSession.cpp
@@ -1,3 +1,4 @@
+#include "FBS/srtpParameters_generated.h"
 #define MS_CLASS "RTC::SrtpSession"
 // #define MS_LOG_DEV_LEVEL 3
 
@@ -28,6 +29,43 @@ namespace RTC
 		}
 	}
 
+	FBS::SrtpParameters::SrtpCryptoSuite SrtpSession::CryptoSuiteToFbs(CryptoSuite cryptoSuite)
+	{
+		switch (cryptoSuite)
+		{
+			case SrtpSession::CryptoSuite::AEAD_AES_256_GCM:
+				return FBS::SrtpParameters::SrtpCryptoSuite::AEAD_AES_256_GCM;
+
+			case SrtpSession::CryptoSuite::AEAD_AES_128_GCM:
+				return FBS::SrtpParameters::SrtpCryptoSuite::AEAD_AES_128_GCM;
+
+			case SrtpSession::CryptoSuite::AES_CM_128_HMAC_SHA1_80:
+				return FBS::SrtpParameters::SrtpCryptoSuite::AES_CM_128_HMAC_SHA1_80;
+
+			case SrtpSession::CryptoSuite::AES_CM_128_HMAC_SHA1_32:
+				return FBS::SrtpParameters::SrtpCryptoSuite::AES_CM_128_HMAC_SHA1_32;
+		}
+	}
+
+	SrtpSession::CryptoSuite SrtpSession::CryptoSuiteFromFbs(FBS::SrtpParameters::SrtpCryptoSuite cryptoSuite)
+	{
+		switch (cryptoSuite)
+		{
+			case FBS::SrtpParameters::SrtpCryptoSuite::AEAD_AES_256_GCM:
+				return SrtpSession::CryptoSuite::AEAD_AES_256_GCM;
+
+			case FBS::SrtpParameters::SrtpCryptoSuite::AEAD_AES_128_GCM:
+				return SrtpSession::CryptoSuite::AEAD_AES_128_GCM;
+
+			case FBS::SrtpParameters::SrtpCryptoSuite::AES_CM_128_HMAC_SHA1_80:
+				return SrtpSession::CryptoSuite::AES_CM_128_HMAC_SHA1_80;
+
+			case FBS::SrtpParameters::SrtpCryptoSuite::AES_CM_128_HMAC_SHA1_32:
+				return SrtpSession::CryptoSuite::AES_CM_128_HMAC_SHA1_32;
+		}
+
+	}
+
 	void SrtpSession::OnSrtpEvent(srtp_event_data_t* data)
 	{
 		MS_TRACE();
